packages/terra-i18n/src/I18nProvider.jsx

```diff
@@ -18,13 +18,27 @@ const propTypes = {
   messages: PropTypes.object.isRequired,
 };
 
-const I18nProvider = props => (
-  <IntlProvider locale={props.locale} key={props.locale} messages={props.messages}>
-    <div>
-      {props.children}
-    </div>
-  </IntlProvider>
-);
+const I18nProvider = (props) => {
+  const {
+    children,
+    locale,
+    messages,
+    ...customProps
+  } = props;
```

Comment:
These props could be spread in the definition above.

```
const I18nProvider = ({ children, locale, messages, ...customProps }) => (
  <IntlProvider
    {...customProps}
    locale={locale}
    key={locale}
    messages={messages}
  >
    <div>
      {children}
    </div>
  </IntlProvider>
);
```
I believe this is how most of our functional components are handling props